File: src/main/java/imericxu/mazegen/logic/maze_algos/MazeAlgorithm.java
package imericxu.mazegen.logic.maze_algos;

import imericxu.mazegen.logic.Algorithm;
import imericxu.mazegen.logic.Node;
import imericxu.mazegen.logic.State;

import java.util.Arrays;

public abstract class MazeAlgorithm extends Algorithm {
	public MazeAlgorithm(Node[] nodes) {
		super(nodes, new State[nodes.length]);
		Arrays.fill(states, State.EMPTY);
	}
}

File: src/main/java/imericxu/mazegen/core/maze_algorithm/MazeAlgorithm.kt
package imericxu.mazegen.core.maze_algorithm

import imericxu.mazegen.core.Algorithm
import imericxu.mazegen.core.Node
import imericxu.mazegen.core.State

abstract class MazeAlgorithm(nodes: Array<Node>) : Algorithm(nodes) {
	override val states: Array<State> = Array(nodes.size) { State.EMPTY }
}
